tests/specs/integration/BaseEntity/ChildClassSpec.cfc

```diff
@@ -0,0 +1,297 @@
+
+                comment.save();
+                
+                comment = getInstance( "InternalComment" ).findOrFail( comment.getId() );
+
+                var memento = comment.getMemento();
+
+                expect( memento )
+                    .toHaveKey( "id" )
+                    .toHaveKey( "body" )
+                    .toHaveKey( "reason" )
+                    .toHaveKey( "commentableId" )
+                    .toHaveKey( "commentableType" )
+                    .toHaveKey( "createdDate" )
+                    .toHaveKey( "modifiedDate" );
+
+                expect( comment.getId() ).notToBeNull();
+                expect( comment.isLoaded() ).toBeTrue();
+
+                expect( comment.getBody() ).toBe( "Lorem ipsum dolor" );
+                expect( comment.getReason() ).toBe( "This is nonsense and I don't want anyone to see it." );
+
+            } );
+
+            it( "can delete a discriminated entity", function(){
+                var newComment = getInstance( "InternalComment" ).fill(
+                    {
+                        "reason" : "I like to keep things private",
+                        "body" : "Lorem ipsum",
+                        "commentableId" : 1345,
+                        "commentableType" : 'Post',
+                        "userId" : getInstance( "User" ).first().getId(),
+                        "createdDate" : now(),
+                        "modifiedDate" : now()
+                    }
+                ).save();
+
+                transactionSetSavePoint( 'comment-saved' );
+
+                getInstance( "InternalComment" ).findOrFail( newComment.getId() ).delete();
+
+                transactionSetSavePoint( 'comment-deleted' );
+
+                expect( isNull( getInstance( "InternalComment" ).find( newComment.getId() ) ) ).toBeTrue();
+
+            });
+
+            it( "Will return a child entity when retrieving a single entity from the parent", function(){
+                var newComment = getInstance( "InternalComment" ).fill(
+                    {
+                        "reason" : "I like to keep things private",
+                        "body" : "Lorem ipsum",
+                        "commentableId" : 1345,
+                        "commentableType" : 'Post',
+                        "userId" : getInstance( "User" ).first().getId(),
+                        "createdDate" : now(),
+                        "modifiedDate" : now()
+                    }
+                ).save();
+
+                var parentRetrieval = getInstance( "Comment" ).find( newComment.getId() );
+                expect( isInstanceOf( parentRetrieval, "InternalComment" ) ).toBeTrue();
+            });
+
+            it( "Will return an array of child classes when fetching multiple results from the parent class", function(){
+                var internalComments = getInstance( "Comment" ).where( 'designation', 'internal' ).get();
+                internalComments.each( function( comment ){
```

Comment:
Yes, it will give you a mix of classes, if you just did an `all()`.  In this case, you would get some `Comment` entities and 1 `InternalComment` entity.